MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -738,19 +814,22 @@ public News getNews() {
         return news;
     }
 
+    private int getLastForceId() {
+        return forceIds.size() > 0 ? forceIds.lastKey() : 0;
```

Comment:
Does this really offer an advantage? Instead of tracking the last index and incrementing each time something is added, you're requiring a recursive search of the map each time to find the last entry (best case log2 n, worst case n). Tracking the next legal value and incrementing it each time is a fairly common practice. The same goes for parts, missions, and scenarios.

Reply:
This code should not have come over as part of this work.